docs/user-guide/gpudeduplication.rst

```diff
@@ -5,164 +5,367 @@
-                   --output-dir /path/to/dedup_output \
-                   --ngram-size char_ngram_size_for_similarity \
-                   # --scheduler-file /path/to/file.json
+               # same as `python jaccard_compute.py`
+               jaccard_compute \
+                 --shuffled-docs-path /path/to/dedup_output/shuffled_docs.parquet \
+                 --output-dir /path/to/dedup_output \
+                 --ngram-size char_ngram_size_for_similarity \
+                 # --scheduler-file /path/to/file.json
 
-  6. Connected Components
-    - Input: jaccard_similarity_results.parquet
-    - Output: connected_components.parquet
+.. _fuzzydup_nofp:
+
+3. Skipping the false positive check (more performant). Not needed if the false positive check was performed.
+
+  a. Buckets to Edgelist
+    - Input: ``_buckets.parquet``
+    - Output: ``_edges.parquet``
     - Example call:
 
-         .. code-block:: bash
+       .. code-block:: bash
 
-                 # same as `python connected_components.py`
-                 gpu_connected_component \
-                   --jaccard-pairs-path /path/to/dedup_output/jaccard_similarity_results.parquet \
-                   --output-dir /path/to/dedup_output \
-                   --cache-dir /path/to/cc_cache \
-                   --jaccard-threshold 0.8
-                   # --scheduler-file /path/to/file.json
+               python buckets_to_edges.py \
+                 --input-bucket-dir /path/to/dedup_output/_buckets.parquet \
+                 --output-dir /path/to/dedup_output \
+                 --input-json-id-field id_column_name
+                 # --scheduler-file /path/to/file.json
+
+4. Connected Components
+  - Input: ``jaccard_similarity_results.parquet``
+  - Output: ``connected_components.parquet``
+  - Example call:
+
+       .. code-block:: bash
+
+               # same as `python connected_components.py`
+               gpu_connected_component \
+                 --jaccard-pairs-path /path/to/dedup_output/jaccard_similarity_results.parquet \
+                 --output-dir /path/to/dedup_output \
+                 --cache-dir /path/to/cc_cache \
+                 --jaccard-threshold 0.8
+                 # --scheduler-file /path/to/file.json
+
+.. caution::
+  The CLI utilities are limited to Jsonl datasets, only work with specific ID formats.
+  For different dataset format use the :ref:`fuzzydup_pyapi`.
+
+------------------------
+Incremental Fuzzy Dedup
+------------------------
 
-* Incremental Fuzzy Dedup
-    To incrementally perform fuzzy dedup, organize your incremental dataset snapshots into separate directories and pass a list of all your directories to :code:`gpu_compute_minhashes`. All other subsequent steps can be done as described above without modification.
+* To incrementally perform fuzzy dedup we do not need to recompute minhashes for datasets where minhashes were already computed.
```

Comment:
```suggestion
* To incrementally perform fuzzy deduplication, we do not need to recompute minhashes for datasets where minhashes were already computed.
```
Can you elaborate more on what "incrementally performing fuzzy deduplication" means?

Reply:
It refers to how fuzzy dedup must be performed if any new data is added to the corpus. For some stages like classification, we just need to apply to the new data. For deduplication, we don't need to recompute min hashes, but all subsequent steps need to be run on this new data in addition to the old datasets.